game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java

```diff
@@ -223,44 +224,43 @@ public AggregateResults calculate(
+      final int runCount)
       throws IllegalStateException {
     synchronized (mutexCalcIsRunning) {
       awaitLatch();
       final long start = System.currentTimeMillis();
+      final AtomicBoolean overflowRunsAvailable = new AtomicBoolean(true);
       // Create worker thread pool and start all workers
-      int totalRunCount = 0;
-      int runCount = initialRunCount;
-      final int workerNum = workers.size();
-      final int workerRunCount = Math.max(1, (runCount / Math.max(1, workerNum)));
-      final List<Future<AggregateResults>> list = new ArrayList<>();
-      for (final BattleCalculator worker : workers) {
-        if (!getIsReady()) {
-          // we could have attempted to set a new game data, while the old one was still being set,
-          // causing it to abort with null data
-          return new AggregateResults(0);
-        }
-        final int currentWorkedRunCount = (runCount <= 0 ? 0 : workerRunCount);
-        if (currentWorkedRunCount > 0) {
-          totalRunCount += currentWorkedRunCount;
-          list.add(
-              executor.submit(
-                  () ->
-                      worker.calculate(
-                          attacker,
-                          defender,
-                          location,
-                          attacking,
-                          defending,
-                          bombarding,
-                          territoryEffects,
-                          retreatWhenOnlyAirLeft,
-                          currentWorkedRunCount)));
-        }
-        runCount -= workerRunCount;
+      final int runsPerWorker = runCount / workers.size();
+      final List<Future<AggregateResults>> list =
+          workers.stream()
+              .map(
+                  worker ->
+                      executor.submit(
+                          () ->
+                              worker.calculate(
+                                  attacker,
+                                  defender,
+                                  location,
+                                  attacking,
+                                  defending,
+                                  bombarding,
+                                  territoryEffects,
+                                  retreatWhenOnlyAirLeft,
+                                  // Ensure that we always achieve the target run count even if
+                                  // the number is not dividable by workers.size()
+                                  (overflowRunsAvailable.getAndSet(false)
+                                          ? runCount % workers.size()
+                                          : 0)
+                                      + runsPerWorker)))
+              .collect(Collectors.toList());
+      if (!getIsReady()) {
+        // we could have attempted to set a new game data, while the old one was still being set,
+        // causing it to abort with null data
+        return new AggregateResults(0);
```

Comment:
tbh I'm not quite sure if we actually need this.
If something is wrong with the data we should get an exception at some point, but IMO this doesn't add anything to potential error handling